x-pack/plugins/security_solution/public/timelines/containers/local_storage/index.tsx

```diff
@@ -43,84 +43,71 @@ export const migrateColumnLabelToDisplayAsText = (
-  const allTimelines = storage.get(LOCAL_STORAGE_TIMELINE_KEY);
+export const getDataTablesInStorageByIds = (storage: Storage, tableIds: TableIdLiteral[]) => {
+  const allDataTables = storage.get(LOCAL_STORAGE_TABLE_KEY);
 
-  if (!allTimelines) {
-    return EMPTY_TIMELINE;
+  if (!allDataTables) {
+    return EMPTY_TABLE;
   }
 
-  return timelineIds.reduce((acc, timelineId) => {
-    const timelineModel = allTimelines[timelineId];
-    if (!timelineModel) {
+  return tableIds.reduce((acc, tableId) => {
+    const tableModel = allDataTables[tableId];
+    if (!tableModel) {
       return {
         ...acc,
       };
     }
 
     return {
       ...acc,
-      [timelineId]: {
-        ...timelineModel,
-        ...(timelineModel.sort != null && !Array.isArray(timelineModel.sort)
-          ? { sort: [timelineModel.sort] }
+      [tableId]: {
+        ...tableModel,
+        ...(tableModel.sort != null && !Array.isArray(tableModel.sort)
+          ? { sort: [tableModel.sort] }
           : {}),
-        ...(Array.isArray(timelineModel.columns)
+        ...(Array.isArray(tableModel.columns)
           ? {
-              columns: timelineModel.columns
+              columns: tableModel.columns
                 .map(migrateColumnWidthToInitialWidth)
                 .map(migrateColumnLabelToDisplayAsText),
             }
           : {}),
       },
     };
-  }, {} as { [K in TimelineIdLiteral]: TimelineModel });
+  }, {} as { [K in TableIdLiteral]: TGridModel });
 };
 
-export const getAllTimelinesInStorage = (storage: Storage) =>
-  storage.get(LOCAL_STORAGE_TIMELINE_KEY) ?? {};
-
-export const addTimelineInStorage = (
-  storage: Storage,
-  id: TimelineIdLiteral,
-  timeline: TimelineModel
-) => {
-  const timelineToStore = cleanStorageTimeline(timeline);
-  const timelines = getAllTimelinesInStorage(storage);
-  storage.set(LOCAL_STORAGE_TIMELINE_KEY, {
-    ...timelines,
-    [id]: timelineToStore,
+export const getAllDataTablesInStorage = (storage: Storage) =>
+  storage.get(LOCAL_STORAGE_TABLE_KEY) ?? {};
+
+export const addTableInStorage = (storage: Storage, id: TableIdLiteral, table: TGridModel) => {
+  const tableToStore = cleanStorageDataTable(table);
+  const tables = getAllDataTablesInStorage(storage);
+  storage.set(LOCAL_STORAGE_TABLE_KEY, {
+    ...tables,
+    [id]: tableToStore,
   });
 };
 
-const cleanStorageTimeline = (timeline: TimelineModel) => {
+const cleanStorageDataTable = (table: TGridModel) => {
   // discard unneeded fields to make sure the object serialization works
-  const {
-    documentType,
-    filterManager,
-    isLoading,
-    loadingText,
-    queryFields,
-    selectAll,
-    unit,
-    ...timelineToStore
-  } = timeline;
-  return timelineToStore;
+  const { isLoading, loadingText, queryFields, unit, ...tableToStore } = table;
+  return tableToStore;
 };
 
-export const useTimelinesStorage = (): TimelinesStorage => {
+export const useDataTablesStorage = (): DataTablesStorage => {
```

Comment:
We would most likely want to handle the migration here from `timelineId` to `dataTable`. I'm not too sure about the rxjs`epicTimeline` either, but I think you _should_ only need to make changes here